src/history/hs_rec.c

```diff
@@ -680,16 +646,25 @@ __wt_hs_insert_updates(
 }
 
 /*
- * __wt_hs_delete_key_from_ts --
- *     Delete history store content of a given key from a timestamp.
+ * __wt_hs_delete_reinsert_key_from_ts --
+ *     Delete history store content of a given key from a timestamp and optionally reinsert them
+ *     with ts-1 timestamp.
```

Comment:
I believe we are bounding the scope of this function by only reinserting records at ts-1. Can we explicitly provide the reinsert timestamp to the function?

Reply:
I don't want to give too much freedom to the caller at the moment. If such use case appear, we can do that.